File: advertisingonrails/app/models/creativeasset.rb

class CreativeAsset < ApplicationRecord
  enum CreativeType: [:Image, :Video, :HTML5, :Audio]
  enum AdFormat: [:Banner, :Video, :Native, :Audio, :Interstitial, :RichMedia, :SearchText, :SocialPost, :CTVVideo]


  composed_of :uRL,
    class_name: "URL",
    mapping: [
      %w[uRL_href href]
    ]

  composed_of :uRL,
    class_name: "URL",
    mapping: [
      %w[uRL_href href]
    ]

  has_many :Files, class_name: 'CreativeFile'
  has_many :Approvals, class_name: 'CreativeApproval'
  has_many :Variations, class_name: 'CreativeVariation'
  has_many :LineItems, class_name: 'LineItem'

end

File: advertisingonaspdotnet/Domain/CreativeAsset.cs

using advertisingonaspdotnet.Contracts;

namespace advertisingonaspdotnet.Domain;

public class CreativeAsset
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? CreativeassetId { get; set; } 
 public virtual string? Name { get; set; } 
 public virtual URL? ClickUrl { get; set; } 
 public virtual URL? LandingPage { get; set; } 
 public virtual int? Width { get; set; } 
 public virtual int? Height { get; set; } 
 public virtual int? DurationSeconds { get; set; } 
public virtual ICollection<CreativeFile> Files { get; set; } = new List<CreativeFile>();
public virtual ICollection<CreativeApproval> Approvals { get; set; } = new List<CreativeApproval>();
public virtual ICollection<CreativeVariation> Variations { get; set; } = new List<CreativeVariation>();
public virtual ICollection<LineItem> LineItems { get; set; } = new List<LineItem>();
 public virtual CreativeType? CreativeType { get; set; } 
 public virtual AdFormat? AdFormat { get; set; } 

    public static CreativeAsset FromRequest(CreativeAssetRequest request) {
        return new CreativeAsset {
            Id = request.Id,
            Name = request.Name,
            ClickUrl = request.ClickUrl,
            LandingPage = request.LandingPage,
            Width = request.Width,
            Height = request.Height,
            DurationSeconds = request.DurationSeconds,
            CreativeType = request.CreativeType,
            AdFormat = request.AdFormat,
        };
    }
}
